File: 2st/Riyeon/소인수분해.py
def solution(n):
    answer = set()
    for i in range(2, n+1):
        if n%i==0:
            cnt=0
            for j in range(2, i+1):
                if i%j == 0:
                    cnt+=1
            if cnt ==1:
                answer.add(i)
    answer = list(answer)
    answer.sort()
    return answer

File: 3st/Jaehee/소인수분해.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public int[] solution(int n) {
        List<Integer> factors = new ArrayList<>();
        
        for (int i = 2; i <= n; i++) {
            if (n % i == 0) {
                factors.add(i);  
                
                while (n % i == 0) {
                    n /= i; 
                }
            }
        }
        
        int[] answer = new int[factors.size()];
        for (int i = 0; i < factors.size(); i++) {
            answer[i] = factors.get(i);
        }
        
        return answer;
    }
}